build/webvr-polyfill.js

```diff
@@ -1269,7 +1269,51 @@ VRDisplay.prototype.requestPresent = function(layers) {
       return;
     }
 
-    self.layer_ = layers[0];
+    var incomingLayer = layers[0];
+    if (!incomingLayer.source) {
+      /*
+      todo: figure out the correct behavior if the source is not provided.
+      see https://github.com/w3c/webvr/issues/58
+      */
+      resolve();
+      return;
+    }
+
+    var leftBounds = incomingLayer.leftBounds || defaultLeftBounds;
+    var rightBounds = incomingLayer.rightBounds || defaultRightBounds;
+    if (wasPresenting) {
+      // Already presenting, just changing configuration
+      var changed = false;
+      var layer = self.layer_;
+      if (layer.source !== incomingLayer.source) {
+        layer.source = incomingLayer.source;
+        changed = true;
+      }
+
+      for (var i = 0; i < 4; i++) {
+        if (layer.leftBounds[i] !== leftBounds[i]) {
+          layer.leftBounds[i] = leftBounds[i];
+          changed = true;
+        }
+        if (layer.rightBounds[i] !== rightBounds[i]) {
+          layer.rightBounds[i] = rightBounds[i];
+          changed = true;
+        }
+      }
+
+      if (changed) {
+        self.fireVRDisplayPresentChange_();
```

Comment:
This is how the browser currently behaves, so I'm copying that. Although, I don't know for sure if the browser fires the event when `requestPresent` has been called with all identical values - I just did it here to err on the side of efficiency.

But the browser definitely does currently fire the event when those values change, which is why my example is able to work at all. Even we decide not to fire `vrdisplaypresentchange`, there should be some other event to determine when the parameters change. I don't want to have to query these values on every single frame.

For reference: https://github.com/w3c/webvr/issues/23